cli/azd/pkg/infra/provisioning/deployment.go

```diff
@@ -50,3 +56,25 @@ func (p *InputParameter) HasValue() bool {
 func (p *InputParameter) HasDefaultValue() bool {
 	return p.DefaultValue != nil
 }
+
+func (p *InputParameter) SetValue(value string) error {
```

Comment:
Would have have a slight preference if instead of modifying existing `InputParameter` values, we just created new instances when we needed to do replacements.  I think it is easier to reason about the system when more stuff is immutable.

At some point, we'll probably also want to support using `--parameter` to assign values to both object and array parameters (in this case, I expect we'd ask the user to pass the JSON encoding of the object or array they want to assign on the command line).

Reply:
This func ensures that any data conversion is done such that `Value interface{}` has the appropriate value type, which is necessary downstream. If we just create a new `InputParameter`, we have to do that at the call site which isn't a good separation of responsibilities.